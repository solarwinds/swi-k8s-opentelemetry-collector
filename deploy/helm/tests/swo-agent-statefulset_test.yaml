# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: Test for swo-agent-statefulset
templates:
  - swo-agent-statefulset.yaml
tests:
  - it: Image should be correct in default state
    template: swo-agent-statefulset.yaml
    set:
      swoagent.enabled: true
    asserts:
    - equal:
          path: spec.template.spec.containers[0].image
          value: solarwinds/swo-agent:v2.8.90
  - it: Image should be correct when overriden tag
    template: swo-agent-statefulset.yaml
    set:
      swoagent.enabled: true
      swoagent.image.tag: "beta1"
    asserts:
    - equal:
          path: spec.template.spec.containers[0].image
          value: solarwinds/swo-agent:beta1
  - it: Image should be correct when overriden by azure
    template: swo-agent-statefulset.yaml
    set:
      swoagent.enabled: true
      aks: true
      global.azure.images.swoagent.digest: "abcd"
      global.azure.images.swoagent.image: "swo-agent:v1.2.3"
      global.azure.images.swoagent.registry: "azurek8s.azure.io/marketplaceimages"
    asserts:
    - equal:
          path: spec.template.spec.containers[0].image
          value: azurek8s.azure.io/marketplaceimages/swo-agent:v1.2.3@abcd
