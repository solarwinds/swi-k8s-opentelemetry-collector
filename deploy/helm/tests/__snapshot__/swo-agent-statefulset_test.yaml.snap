SWO Agent spec should match snapshot when swoagent is enabled:
  1: |
    affinity:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
            - matchExpressions:
                - key: kubernetes.io/arch
                  operator: In
                  values:
                    - amd64
                    - arm64
                - key: kubernetes.io/os
                  operator: In
                  values:
                    - linux
    containers:
      - env:
          - name: UAMS_CLIENT_ID_OVERRIDE_SOURCE_NAME
            valueFrom:
              configMapKeyRef:
                key: CLUSTER_UID
                name: RELEASE-NAME-swo-k8s-collector-common-env
          - name: SWO_URL
            value: <OTEL_ENVOY_ADDRESS>
          - name: UAMS_ACCESS_TOKEN
            valueFrom:
              secretKeyRef:
                key: SOLARWINDS_API_TOKEN
                name: solarwinds-api-token
          - name: CLUSTER_UID
            valueFrom:
              configMapKeyRef:
                key: CLUSTER_UID
                name: RELEASE-NAME-swo-k8s-collector-common-env
          - name: CLUSTER_NAME
            valueFrom:
              configMapKeyRef:
                key: CLUSTER_NAME
                name: RELEASE-NAME-swo-k8s-collector-common-env
          - name: MANIFEST_VERSION
            valueFrom:
              configMapKeyRef:
                key: MANIFEST_VERSION
                name: RELEASE-NAME-swo-k8s-collector-common-env
          - name: HTTPS_PROXY
            valueFrom:
              configMapKeyRef:
                key: HTTPS_PROXY
                name: RELEASE-NAME-swo-k8s-collector-common-env
                optional: true
          - name: UAMS_IS_EPHEMERAL
            value: "true"
        image: solarwinds/swo-agent:v2.10.333
        imagePullPolicy: IfNotPresent
        name: swo-agent
        resources:
          limits:
            memory: 800Mi
          requests:
            cpu: 100m
            memory: 800Mi
        volumeMounts:
          - mountPath: /uamsclient/workdir
            name: uams-client-workdir
    securityContext: {}
    serviceAccountName: RELEASE-NAME-swo-k8s-collector
    volumes:
      - name: uams-client-workdir
        persistentVolumeClaim:
          claimName: RELEASE-NAME-swo-k8s-collector-swo-agent
